uxarray/grid/grid.py

```diff
@@ -231,6 +231,24 @@ def from_dataset(
 
         return cls(grid_ds, source_grid_spec, source_dims_dict)
 
+    @classmethod
+    def from_geodataframe(
```

Comment:
```suggestion
    def from_geometry_file()
```

This signature might be more appropriate, since we aren't explicity passing in a GeoDataFrame to this function. Instead, it's a file containing the geometry in a format that is then read in as a GeoDataFrame. 

This signature might be more appropriate, since we aren't passing in a GeoDataFrame at this step, instead it's a file

Reply:
`from_geometry_file` is too generic - that would mean any geometry, quick one `from_geopandas_geometry`?

Reply:
> Good point.
> 
> Can we instead create a `Grid.from_file()`, which can branch either to Xarray or GeoPandas. It can take a parameter `backend` which can either be `xarray` or `geopandas` for reading a geometry file.
> 
> For reference: https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.from_file.html
> 
> Any other ideas @erogluorhan

I like a more generic` from_file` with a backend; xarray doesn't have a from_file, it does have a `from_dict` which we don't need for unstructured grid. @erogluorhan should I make this change?